asllib/AST.mli

```diff
@@ -143,10 +143,10 @@ and pattern =
   | Pattern_Single of expr
   | Pattern_Tuple of pattern list
 
-(** Indexes an array, a bitvector. *)
+(** Indexes an array, a bitvector, an argument to a getter/setter *)
 and slice =
-  | Slice_Single of expr
-      (** [Slice_Single i] is the slice of length [1] at position [i]. *)
+  | Slice_Arg of expr
```

Comment:
Since this is always a getter argument perhaps we should not even consider it a slice but rather a separate type of AST node? Something like
`getter_arg of expr`?
This would avoid having to fail in matching slices.

Reply:
Because the AST is shared between ASLv0 and ASLv1, the same node has different meanings for the different parsing process. In ASLv1, `Slice_Arg` is used for getter/setter arguments and array indexing. In ASLv0, *in addition to* those, it's also used for slicing a single bit. The aim of this change is to disambiguate some bracket usages in ASLv1 and therefore this distinction is expected.

As for the name, I can't think of other names that can capture the above meaning. I do want to avoid use `Slice_Single` again, because the reader might be confused of the complete different usage in outdated documents.